pkg/client/patch.go

```diff
@@ -211,3 +212,34 @@ func (p applyPatch) Data(obj Object) ([]byte, error) {
 	// client-go does, more-or-less).
 	return json.Marshal(obj)
 }
+
+//applyFromPatch uses server-side-apply with a ApplyConfiguration object.
+type applyFromPatch struct {
+	patch interface{}
+}
+
+// Type implements Patch.
+func (p applyFromPatch) Type() types.PatchType {
+	return types.ApplyPatchType
+}
+
+// Sets the name and namespace for the returned object if an ApplyConfiguration was supplied
+func setNameNamespace(obj Object, ac interface{}) {
+	obj.SetName(reflect.ValueOf(ac).Elem().FieldByName("Name").Elem().String())
```

Comment:
These reflections calls seem insufficiently guarded. `ApplyFrom` takes an `interface{}` and we don't check anywhere if the fields we expect to access here are actually present. Maybe perform the assertions in `ApplyFrom`?

Reply:
Yeah I agree. I wasn't sure if this was the correct approach so most of the error checking wasn't included. Will improve the handling for errors and edge cases.